### ISSUE TITLE
`Shift+Delete` (cut) not putting the deleted text onto clipboard

### ISSUE THREAD

**Issue by breki** [RickStrahl/MarkdownMonster#286]
`Shift+Delete` is a standard alternative keyboard shortcut for cutting (like `Ctrl+X`), but in MarkdownMonster it just deletes the text without putting it onto the clipboard, which is quite annoying if you're used to using this.

**Comment by RickStrahl** [comment]
A number of common text editors use `Shift-Del` for line delete operations including Visual Studio and Visual Studio Code. This was added as an explicitly requested feature actually by quite a few people.

**Comment by breki** [comment]
Hmm that's strange, all of the text editors I use - VS, Notepad, Notepad++,
Word, LibreOffice Writer, text boxes in browser (even the GMail one I'm
currently writing the response in) use Shift+Del as a cut operation and
_not_ line delete.

[image: Inline image 1]

Source: https://msdn.microsoft.com/en-us/library/da5kh0wa.aspx

Visual Studio provides other shortcuts for deleting line (Ctrl+L which also
copies it into the clipboard and Shift+Ctrl+L which doesn't). This same
shortcuts work in Notepad++.

Shift+Del, Ctrl+Ins and Shift+Ins are really ancient shortcuts on Windows
for cutting, copying and pasting.

My guess is the people who requested this non-standard behavior belong to
the same group as these ones (probably being used to some other editor
which does not follow the standard Windows hotkeys scheme):

   -
   https://stackoverflow.com/questions/7745119/disable-shiftdelete-cutting-in-visual-studio
   -
   https://stackoverflow.com/questions/46220035/sublime-text-3-disable-shift-delete-copying-the-selected-text

**Comment by RickStrahl** [comment]
Hmmm... not sure. I'm using the default key configuration in both VS and VS Code and both delete a line with Shift-Del - when there is no selection.

There was a fix in MM for a bug where Shift-Delete always deleted even when there was a selection, which was a bug. That was fixed I think in 1.8.5. If not check 1.8.8 which I just put out.

Come to think of it, I'm sure that's what you're running into because if a selection is active, just the selection is cut, not the entire line.

+++ Rick ---

**Comment by breki** [comment]
OK, I get it now: `Shift+Delete` has two modes of function (in Visual Studio and other text editors I mentioned previously):
1. when there is a selected text, it functions as cut text (the same as `Ctrl+X`)
2. when there is no text selection, it deletes the line - TBH I never knew about this one, I got used to `Ctrl+L` and `Shift+Ctrl+L`

I've tried it in 1.8.8 and the mode 1 still doesn't work properly. Yes, it deletes the selected text, but it doesn't save the deleted text into the clipboard. And this is my main gripe, because when I then try to paste the cut text somewhere else, it pastes something completely different (the previous contents of the clipboard).
